tests/test_ai/test_priority_algorithm/test_mutations.py

```diff
@@ -0,0 +1,263 @@
+
+        self.assertEqual(
+            get_priority_team(0, priority_team_set).student_ids,
+            get_priority_team(0, mutated_team_set).student_ids,
+            "Locked teams should not change",
+        )
+
+    def test_mutate_robinhood__does_not_make_team_set_worse(self):
+        """
+        The score of the mutated team set should be at least as good as the original team set
+        """
+        priority_team_set, student_dict = create_new_priority_team_set(3, 9)
+        priorities = [TestPriority()]
+
+        mutated_team_set = mutate_robinhood(priority_team_set, priorities, student_dict)
+
+        # Reset the scores to force them to be recalculated
+        priority_team_set.score = None
+        mutated_team_set.score = None
+
+        self.assertGreaterEqual(
+            mutated_team_set.calculate_score(priorities, student_dict),
+            priority_team_set.calculate_score(priorities, student_dict),
+            "The score of the mutated team set should be at least as good as the original team set",
+        )
+
+        # Also try this when the original team set is optimal
+        priority_team_set, student_dict = create_new_priority_team_set(3, 9)
+        priority_team_set.priority_teams[0].student_ids = [0, 1, 2]
+        priority_team_set.priority_teams[1].student_ids = [3, 4, 5]
+        priority_team_set.priority_teams[2].student_ids = [6, 7, 8]
+
+        mutated_team_set = mutate_robinhood(priority_team_set, priorities, student_dict)
+
+        # Reset the scores to force them to be recalculated
+        priority_team_set.score = None
+        mutated_team_set.score = None
+
+        self.assertGreaterEqual(
+            mutated_team_set.calculate_score(priorities, student_dict),
+            priority_team_set.calculate_score(priorities, student_dict),
+            "The score of the mutated team set should be at least as good as the original team set",
+        )
+
+    def test_mutate_robinhood__all_students_in_mutated_team_set(self):
+        """
+        All students in the original team set should be in the mutated team set
+        """
+        priority_team_set, student_dict = create_new_priority_team_set(3, 9)
+        priorities = [TestPriority()]
+
+        mutated_team_set = mutate_robinhood(priority_team_set, priorities, student_dict)
+
+        # Count the number of students in the original team set that are not in the mutated team set
+        all_students = [
+            student.id
+            for priority_team in priority_team_set.priority_teams
+            for student in priority_team.team.students
+        ]
+
+        assert len(all_students) == 9
+
+        missing_students = 0
+        for i in range(9):
+            if i not in all_students:
+                missing_students += 1
+
+        self.assertEqual(
+            missing_students, 0, "Not all students are in the mutated team set"
+        )
+
+    def test_mutate_robinhood__handles_invalid_arguments(self):
```

Comment:
note;
yikes my bad, looks like this test is juicy but I don't think mutations should really be doing any "validation" at all, thus no such thing as handling invalid arguments